sql: extend TestLogic to ease investigation of failures.

knz opened the pull request:
Summary of changes:
- TestLogic now can make extra efforts to continue testing after it
  encounters the first error when `-try-harder` is specified. This way, one can collect multiple
  errors to investigate during a single run of TestLogic.
- TestLogic now makes extra effort to report the execution progress of
  large tests. For this:
  - The progress indicator every two seconds is fixed for long tests
    involving many small input files.
  - The printing of progress with `-v` is now decoupled from the
    printing of the SQL strings sent to the databse, which are now
    only printed if `-show-sql` is passed.
- A new flag `-flex-types` is introduced to tolerate mismatches
  between the requested numeric types of result column and the actual
  type returned by a query. If this flag is passed and a query returns
  a different numeric type than expected (but both expected and
  provided types are numeric), and provided the other test conditions
  do hold (expected number of rows, hash, etc) the error is ignored
  and the test is counted as passed.
- A new flag `-soft` is introduced to tolerate queries that are not
  recognized by the SQL dialect; that is queries for which the
  preparation of the query plan fails unexpectedly. This is meant to
  ignore tests for which CockroachDB does not support the input
  syntax.
- For both `-soft` and `-flex-types`, a summary of the queries whose
  errors have been ignored is printed at the end when `-v -error-summary` is also
  given. This summary groups ignored queries by error message.
  - By default, the text of the queries and the text of the error
    messages in this summary is shortened to fit on a medium-sized
    terminal. To output the full text, one can use the option
    `-full-summary`.

For example, by using both `-soft -flex-types`, most expressions from
sqllogictest's `random/expr` tree now pass. This means that each test
expression is either clearly recognized as unsupported during prepare,
or it is recognized as supported and its evaluation succeeds with a
type compatible with the expected type.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7409)

<!-- Reviewable:end -->

knz committed:
sql: extend TestLogic to ease investigation of failures.

Summary of changes:

- TestLogic now can make extra efforts to continue testing after it
  encounters the first error, if `-try-harder` is specified. This way,
  one can collect multiple errors to investigate during a single run
  of TestLogic.

- TestLogic now makes extra effort to report the execution progress of
  large tests. For this:

  - The progress indicator every two seconds is fixed for long tests
    involving many small input files.

  - The printing of progress with `-v` is now decoupled from the
    printing of the SQL strings sent to the databse, which are now
    only printed if `-show-sql` is passed.

- A new flag `-flex-types` is introduced to tolerate mismatches
  between the requested numeric types of result column and the actual
  type returned by a query. If this flag is passed and a query returns
  a different numeric type than expected (but both expected and
  provided types are numeric), and provided the other test conditions
  do hold (expected number of rows, hash, etc) the error is ignored
  and the test is counted as passed.

- A new flag `-soft` is introduced to tolerate queries that are not
  recognized by the SQL dialect; that is queries for which the
  preparation of the query plan fails unexpectedly. This is meant to
  ignore tests for which CockroachDB does not support the input
  syntax.

- For both `-soft` and `-flex-types`, a summary of the queries whose
  errors have been ignored can be printed at the end when `-v` and
  `-error-summary` are also given. This summary groups ignored queries
  by error message.

  - By default, the text of the queries and the text of the error
    messages in this summary is shortened to fit on a medium-sized
    terminal. To output the full text, one can use the option
    `-full-messages`.

For example, by using both `-soft -flex-types`, most expressions from
sqllogictest's `random/expr` tree now pass. This means that each test
expression is either clearly recognized as unsupported during prepare,
or it is recognized as supported and its evaluation succeeds with a
type compatible with the expected type.

knz commented:
cc @dt @paperstreet @nvanbenschoten @RaduBerinde @tamird @petermattis @mjibson  :)

knz commented:
(needed to investigate #7351)

dt commented:
Two high level comments:
1) let's break this up into separate PRs.
2) I think we should swap the default on `-stop-early` since keeping the error at the end is nice when.

RaduBerinde commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Great stuff. I agree with @dt we should just `stop-early` by default (I think the cases where one wastes time finding the error outnumber the cases where multiple errors help).

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[sql/logic_test.go, line 61 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KKu8Wpqu4uIKOb_I3-x:-KKu8Wpqu4uIKOb_I3-y:-610429217) ([raw file](https://github.com/cockroachdb/cockroach/blob/9a52df11c27916b541d3d7abe887d5e00bbb55f3/sql/logic_test.go#L61)):*
> ```Go
> 	maxErrsPerInput = flag.Int("max-errors", 10, "stop processing each input file after this number of errors (set to 0 for no limit)")
> 	softTest        = flag.Bool("soft", false, "tolerate unexpected errors when preparing a query")
> 	showSQL         = flag.Bool("show-sql", false, "print the individual SQL strings before processing")
> ```

What are "SQL strings"? Maybe SQL statements?

---

*[sql/logic_test.go, line 188 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KKu8cbBjAuN9D75BNwg:-KKu8cbBjAuN9D75BNwh:2021056286) ([raw file](https://github.com/cockroachdb/cockroach/blob/9a52df11c27916b541d3d7abe887d5e00bbb55f3/sql/logic_test.go#L188)):*
> ```Go
> }
> 
> func shortenString(msg string) string {
> ```

function needs a comment (it's not obvious what it's doing without reading the code)

---

*[sql/logic_test.go, line 983 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KKu91CmoAlVNW0KhHLw:-KKu91CmoAlVNW0KhHLx:-1248221774) ([raw file](https://github.com/cockroachdb/cockroach/blob/9a52df11c27916b541d3d7abe887d5e00bbb55f3/sql/logic_test.go#L983)):*
> ```Go
> 		if now.Sub(lastProgress) >= 2*time.Second {
> 			lastProgress = now
> 			fmt.Fprintf(os.Stderr, "--- total progress: %d statements/queries\n", total)
> ```

Seems odd that this progress is on stderr when the rest (e.g. the "total:") is not

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-:-KKu9nypIz17Bt3uQmug:2125117505)*
<!-- Sent from Reviewable.io -->

knz commented:
I have replaced `-stop-early` with `-try-harder` to invert the semantics of the flag. Good call.

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[sql/logic_test.go, line 61 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KKu8Wpqu4uIKOb_I3-x:-KL3pfzcnvUskHwFvvW1:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/9a52df11c27916b541d3d7abe887d5e00bbb55f3/sql/logic_test.go#L61)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> What are "SQL strings"? Maybe SQL statements?
</details>
Done.

---

*[sql/logic_test.go, line 188 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KKu8cbBjAuN9D75BNwg:-KL3phNdPGScIIm6R2zB:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/9a52df11c27916b541d3d7abe887d5e00bbb55f3/sql/logic_test.go#L188)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> function needs a comment (it's not obvious what it's doing without reading the code)
</details>
Done.

---

*[sql/logic_test.go, line 983 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KKu91CmoAlVNW0KhHLw:-KL3pujxeu_JOu1Fcik8:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/9a52df11c27916b541d3d7abe887d5e00bbb55f3/sql/logic_test.go#L983)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> Seems odd that this progress is on stderr when the rest (e.g. the "total:") is not
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-:-KL3pQlVlQ5HVzd5LVYs:-1470727964)*
<!-- Sent from Reviewable.io -->

knz commented:
@jordanlewis curious to know what you think of this, and if you have any language/wording remarks about the main comment in the source (fresh eyes, etc). Feel free to de-assign/re-assign.

tamird commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 30 unresolved discussions, some commit checks failed.

---

*[sql/logic_test.go, line 57 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXZW3osaZjsaqik_U:-KLHXZW3osaZjsaqik_V:1046727041) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L57)):*
> ```Go
> //
> // TestLogic is used both for direct unit testing of CockroachDB
> // during development cycles, and regular (nightly) stress testing of
> ```

the nightly run is the bigtest, not a stress test.

---

*[sql/logic_test.go, line 72 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXuuLtWPdXeLlISpL:-KLHXuuLtWPdXeLlISpM:1347347953) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L72)):*
> ```Go
> // The test input is expressed using a domain-specific language, called
> // Test-Script, defined by SQLite's "Sqllogictest".  The official home
> // for Sqllogictest and Test-Script is
> ```

s/for/of/

---

*[sql/logic_test.go, line 84 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXyoonpnc3I4m9jKw:-KLHXyopMpMp_j9EAogJ:-1736110928) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L84)):*
> ```Go
> // directives.
> //
> // The Test-Script is line-oriented. It supports both statements which
> ```

s/The//

---

*[sql/logic_test.go, line 99 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYCGN13FodgV4QakD:-KLHYCGN13FodgV4QakE:235881555) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L99)):*
> ```Go
> // - logicTest.processTestFile() runs all tests from that input file.
> //
> // - each test is instantiated either as a logicStatement or
> ```

s/test/line/ since statements are not tests

---

*[sql/logic_test.go, line 106 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYKb_y2Pa2sdgv_W6:-KLHYKb_y2Pa2sdgv_W7:-235107528) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L106)):*
> ```Go
> //
> // - Selection of the input files.
> // - How and how early to stop when tests fail.
> ```

how and when

---

*[sql/logic_test.go, line 107 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYQavhqcb-McRVNvo:-KLHYQavhqcb-McRVNvp:625250311) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L107)):*
> ```Go
> // - Selection of the input files.
> // - How and how early to stop when tests fail.
> // - What gets reported.
> ```

each of these bullets uses a different tense. pick one plz

---

*[sql/logic_test.go, line 131 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYjzT2BZs8IqXiE8m:-KLHYjzT2BZs8IqXiE8n:214220139) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L131)):*
> ```Go
> //                after N errors have been encountered. Default 10.
> //
> // -soft          tolerate / ignore errors that occur during query
> ```

this behaviour is very specific - i don't think soft is a good name for it. perhaps "ignore-unsupported"?

---

*[sql/logic_test.go, line 139 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYtzvxG2KB4nZDxad:-KLHYtzvxG2KB4nZDxae:-1845241054) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L139)):*
> ```Go
> //
> // -flex-types    tolerate when a result column is produced
> //                with another numeric type than the one expected by
> ```

s/another/a different/

---

*[sql/logic_test.go, line 145 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZ7DVosk-ZiZo0z_V:-KLHZ7DVosk-ZiZo0z_W:154535339) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L145)):*
> ```Go
> // What gets reported:
> //
> // -v             be more verbose. This causes database log messages
> ```

this is really test testing package's `-v`, isn't it? I think you should mention that, because in the case of this package's tests being built into a standalone binary the flag becomes `-test.v`.

In fact, I'd replace this comment with "the testing package's verbose flag."

---

*[sql/logic_test.go, line 158 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZTvloXvAEToi0dtU:-KLHZ31g0W9zscJB9eW9:-1177995452) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L158)):*
> ```Go
> //                "FAIL" to indicate an unexpected/undesired test
> //                failure.
> //                Works only in combination with -v.
> ```

remove this - instead, mention that all output is suppressed unless `-v` is passed (or a test fails) in the above comment

---

*[sql/logic_test.go, line 163 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZZABjb13zfKrqxjy:-KLHZZABjb13zfKrqxjz:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L163)):*
> ```Go
> //                of all queries that have caused that error.
> //                Useful with -soft and/or -flex-types.
> //                Works only in combination with -v.
> ```

ditto

---

*[sql/logic_test.go, line 169 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZ_N-UyH1J6Y60c41:-KLHZ_N-UyH1J6Y60c42:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L169)):*
> ```Go
> //                terminal screen. With this parameter, the
> //                full text of errors and queries is printed.
> //                Works only in combination with -v.
> ```

ditto

---

*[sql/logic_test.go, line 186 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZtQ3efZwql2D9oJe:-KLHZtQ3efZwql2D9oJf:876136624) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L186)):*
> ```Go
> 
> 	// Testing mode
> 	tryHarder = flag.Bool("try-harder", false,
> ```

i think you can get rid of this flag completely - just keep max-errors and default it to 1.

---

*[sql/logic_test.go, line 189 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHceqW2nIakVRszyWz:-KLHceqW2nIakVRszyX-:-969471076) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L189)):*
> ```Go
> 		"do not stop the test at the first query error encountered")
> 	maxErrsPerInput = flag.Int("max-errors", 10,
> 		"stop processing each input file after this number of errors (set to 0 for no limit)")
> ```

this name is misleading. I think this flag should control max errors (rather than per file).

---

*[sql/logic_test.go, line 192 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLH_3eqwwm0usqOhyfM:-KLH_3eqwwm0usqOhyfN:1761240811) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L192)):*
> ```Go
> 	softTest  = flag.Bool("soft", false, "tolerate unexpected errors when preparing a query")
> 	flexTypes = flag.Bool("flex-types", false,
> 		"do not fail when a test expects a column of a number type but the query provides another type")
> ```

is this only about numbers?

---

*[sql/logic_test.go, line 239 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLH_f0tOxbGAWnGYG5c:-KLH_f0tOxbGAWnGYG5d:2055050721) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L239)):*
> "" indicates no particular pgwire error code is expected.

This is vague. Does it mean that any pgwire error code is accepted?

---

*[sql/logic_test.go, line 244 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHao5aPy4-52oQjCH3:-KLHao5aPy4-52oQjCH4:-771871300) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L244)):*
> ```Go
> }
> 
> // logicSorter is the common type for result sorter functions.
> ```

remove this line and move the rest of the comment to the fir

---

*[sql/logic_test.go, line 245 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHajHOMti0h0-D6-Ra:-KLHajHOMti0h0-D6-Rb:1317845797) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L245)):*
> ```Go
> 
> // logicSorter is the common type for result sorter functions.
> // Sorting occurs (or not) depending on Test-Script's sorting option
> ```

I think i'd remove the first sen

---

*[sql/logic_test.go, line 303 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHa2EUvkBNWgoPSSbT:-KLHa2EV5244zhagNDyM:-1589215883) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L303)):*
> ```Go
> 	pos string
> 	sql string
> 	// colTypes indicate the expected result column types.
> ```

indicates

---

*[sql/logic_test.go, line 305 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHa4mqIfyDGW7fUx3d:-KLHa4mqIfyDGW7fUx3e:-1187506396) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L305)):*
> ```Go
> 	// colTypes indicate the expected result column types.
> 	colTypes string
> 	// colNames tells to check whether the result columns have the same
> ```

tells to check? reword. perhaps:

colNames controls the inclusion of column names in the query result.

---

*[sql/logic_test.go, line 308 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHabu6tBid_cdpHr5k:-KLHabu6tBid_cdpHr5l:-1745743191) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L308)):*
> ```Go
> 	// names as indicated in the test input.
> 	colNames bool
> 	// label annotates queries so as to check that queries referring to
> ```

hm, don't understand this. do we use it anywhere?

---

*[sql/logic_test.go, line 317 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHb84bYuBRquS-x35a:-KLHb84bYuBRquS-x35b:-565282810) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L317)):*
> ```Go
> 	expectErrCode string
> 	// expectedResults indicate the expected result values, indicated as
> 	// the sequence of output text words that would be produced by
> ```

I'm guessing you're trying to specify the behaviour with respect to whitespace? This reads pretty awkwardly currently, perhaps be more specific.

---

*[sql/logic_test.go, line 321 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHbUunQKTGeEZiVKPr:-KLHbUunQKTGeEZiVKPs:1669118032) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L321)):*
> ```Go
> 	// does not specify row separators.
> 	expectedResults []string
> 	// expectedResultsRaw indicate the same as expectedResults, but
> ```

s/indicate/is/

---

*[sql/logic_test.go, line 329 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHb_bN0XB80RO89RAF:-KLHb_bN0XB80RO89RAG:-145021832) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L329)):*
> ```Go
> 	// combined. "" indicates hash checking is disabled.
> 	expectedHash string
> 	// expectedValues indicate how many rows are expected when
> ```

s/indicate how many rows are/indicates the number of rows/

---

*[sql/logic_test.go, line 366 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHcHikZBB4G-jYdbkp:-KLHcHikZBB4G-jYdbkq:701819826) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L366)):*
> ```Go
> 	// and "traceoff" directives.
> 	traceFile *os.File
> 	// verbose indicate whether -v was passed.
> ```

not true, since its value is `testing.Verbose() || log.V(1)`

---

*[sql/logic_test.go, line 825 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHdJLM-FPXs76A6lcc:-KLHdJLM-FPXs76A6lcd:841390575) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L825)):*
> ```Go
> 	// - error on expected error is worth going further, even
> 	//   if the obtained error does not match the expected error.
> 	if ok := t.verifyError("", stmt.pos, stmt.expectErr, stmt.expectErrCode, err); !ok {
> ```

```
ok := ....
if ok {
  t.finishOne("OK")
}
return ok
```

---

*[sql/logic_test.go, line 832 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHd9aT7Ul3t5khy-Gi:-KLHd9aT7Ul3t5khy-Gj:-694968486) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L832)):*
> ```Go
> }
> 
> func (t *logicTest) execQuery(query logicQuery) {
> ```

there is a ton of `t.Error...return` in this function. Perhaps return an error?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: all files reviewed at latest revision, 30 unresolved discussions, some commit checks failed.

---

*[sql/logic_test.go, line 57 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXZW3osaZjsaqik_U:-KLIuUSYxHOevQQHVWxJ:1289450228) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L57)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> the nightly run is the bigtest, not a stress test.
</details>
Arguably the bigtest is also a stress test. I decided to use "larger-scale" to keep it a bit more general.

---

*[sql/logic_test.go, line 72 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXuuLtWPdXeLlISpL:-KLIuauj_cxy-ce1F1I9:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L72)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/for/of/
</details>
Done.

---

*[sql/logic_test.go, line 84 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXyoonpnc3I4m9jKw:-KLIuPKWGhh8NfP8o31s:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L84)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/The//
</details>
Done.

---

*[sql/logic_test.go, line 99 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYCGN13FodgV4QakD:-KLIugpzMQnkV4mVdp2U:696858999) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L99)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/test/line/ since statements are not tests
</details>
The statements themselves aren't, but the expected output (ok/error) together with the statement is a test. I think it is fair to say that an input file consists of multiple individual tests, which can either exercise statements or queries.

---

*[sql/logic_test.go, line 106 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYKb_y2Pa2sdgv_W6:-KLIv33iSJ34D7qwgVcw:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L106)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> how and when
</details>
Done.

---

*[sql/logic_test.go, line 107 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYQavhqcb-McRVNvo:-KLIvDzQDHDwNJx46wbz:1403920166) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L107)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> each of these bullets uses a different tense. pick one plz
</details>
The tenses were fine, I think the term you meant to use is "semantic category". I see what you mean. :)

---

*[sql/logic_test.go, line 131 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYjzT2BZs8IqXiE8m:-KLIvSto5zlMre3o2ikM:1593290914) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L131)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this behaviour is very specific - i don't think soft is a good name for it. perhaps "ignore-unsupported"?
</details>
Finding a good name for this has eluded me for long. "ignore-unsupported" is inaccurate, because this does not ignore all unsupported queries, just those that fail prepare. The advantage of "soft" is that it is also largely non-descriptive so it does not end up being obviously inaccurate. Feel free to suggest a better name; in the mean time I'll keep "soft".

---

*[sql/logic_test.go, line 139 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYtzvxG2KB4nZDxad:-KLIvvrGUt5Z8OIDhky3:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L139)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/another/a different/
</details>
Done.

---

*[sql/logic_test.go, line 145 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZ7DVosk-ZiZo0z_V:-KLIw88cr12iKUesmrvM:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L145)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this is really test testing package's `-v`, isn't it? I think you should mention that, because in the case of this package's tests being built into a standalone binary the flag becomes `-test.v`.
> 
> In fact, I'd replace this comment with "the testing package's verbose flag."
</details>
Done.

---

*[sql/logic_test.go, line 158 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZTvloXvAEToi0dtU:-KLIwRzLuK3LJWGVbkdI:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L158)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> remove this - instead, mention that all output is suppressed unless `-v` is passed (or a test fails) in the above comment
</details>
Done.

---

*[sql/logic_test.go, line 163 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZZABjb13zfKrqxjy:-KLIwShQQQopS1kpeKV7:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L163)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ditto
</details>
Done.

---

*[sql/logic_test.go, line 169 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZ_N-UyH1J6Y60c41:-KLIwT6HYWs7yAByEhpB:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L169)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ditto
</details>
Done.

---

*[sql/logic_test.go, line 186 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHZtQ3efZwql2D9oJe:-KLIwUK4YWaFtILliC8K:-1643491507) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L186)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> i think you can get rid of this flag completely - just keep max-errors and default it to 1.
</details>
Ok, agreed. Done.

---

*[sql/logic_test.go, line 189 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHceqW2nIakVRszyWz:-KLIzPnjQwgo3Cih0TPT:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L189)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> this name is misleading. I think this flag should control max errors (rather than per file).
</details>
Done.

---

*[sql/logic_test.go, line 192 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLH_3eqwwm0usqOhyfM:-KLIzRPdMLXgxP6ZHb4x:-463013414) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L192)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> is this only about numbers?
</details>
Yes, as explained in the comment above.

---

*[sql/logic_test.go, line 239 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLH_f0tOxbGAWnGYG5c:-KLIzZUfWiWmVmh28qDX:-888365374) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L239)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> > "" indicates no particular pgwire error code is expected.
> 
> This is vague. Does it mean that any pgwire error code is accepted?
</details>
Clarified.

---

*[sql/logic_test.go, line 244 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHao5aPy4-52oQjCH3:-KLIznf0Wt37UrNPLw9O:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L244)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> remove this line and move the rest of the comment to the fir
</details>
Done.

---

*[sql/logic_test.go, line 245 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHajHOMti0h0-D6-Ra:-KLIzo6B792uRZdZ_THD:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L245)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> I think i'd remove the first sen
</details>
Done.

---

*[sql/logic_test.go, line 303 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHa2EUvkBNWgoPSSbT:-KLIzqt6OKjuqi0WTa-r:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L303)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> indicates
</details>
Done.

---

*[sql/logic_test.go, line 305 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHa4mqIfyDGW7fUx3d:-KLIzvoAmD-HE7D8eVFB:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L305)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> tells to check? reword. perhaps:
> 
> colNames controls the inclusion of column names in the query result.
</details>
Done.

---

*[sql/logic_test.go, line 308 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHabu6tBid_cdpHr5k:-KLJ45jVilyvdAxVhqMr:-1556672179) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L308)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> hm, don't understand this. do we use it anywhere?
</details>
You're right we didn't use it so I removed it. However this made me notice that we did not implement Test-Script's label logic at all, so I added that. There's a new member in the logicTest struct for it now.

---

*[sql/logic_test.go, line 317 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHb84bYuBRquS-x35a:-KLJ4mHEsyQqAMblEK-4:1023282425) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L317)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> I'm guessing you're trying to specify the behaviour with respect to whitespace? This reads pretty awkwardly currently, perhaps be more specific.
</details>
I simplified the text altogether. Thanks for the hint.

---

*[sql/logic_test.go, line 321 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHbUunQKTGeEZiVKPr:-KLJ52Qt9TyJT5wiksb2:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L321)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/indicate/is/
</details>
Done.

---

*[sql/logic_test.go, line 329 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHb_bN0XB80RO89RAF:-KLJ56pIxUxLE_-QEn6E:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L329)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> s/indicate how many rows are/indicates the number of rows/
</details>
Done.

---

*[sql/logic_test.go, line 366 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHcHikZBB4G-jYdbkp:-KLJ5MmeCbGe_6pXPSVT:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L366)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> not true, since its value is `testing.Verbose() || log.V(1)`
</details>
Done.

---

*[sql/logic_test.go, line 825 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHdJLM-FPXs76A6lcc:-KLJ5aFEDoCyCZ_ytiET:-932165681) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L825)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> ```
> ok := ....
> if ok {
>   t.finishOne("OK")
> }
> return ok
> ```
</details>
Nice. :) Done.

---

*[sql/logic_test.go, line 832 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHd9aT7Ul3t5khy-Gi:-KLJ6qZlmGr8n0H_xJmw:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L832)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> there is a ton of `t.Error...return` in this function. Perhaps return an error?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks pending.

---

*[sql/logic_test.go, line 57 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXZW3osaZjsaqik_U:-KLJAKAK3yX25M21W9CW:-984648909) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L57)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Arguably the bigtest is also a stress test. I decided to use "larger-scale" to keep it a bit more general.
</details>
sure, you could argue that, but "stress" has a specific meaning in our parlance and refers to the stress utility. large-scale is an overloaded term.

---

*[sql/logic_test.go, line 131 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYjzT2BZs8IqXiE8m:-KLJAy6ScS1jo0F1FVjZ:-1737285665) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L131)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Finding a good name for this has eluded me for long. "ignore-unsupported" is inaccurate, because this does not ignore all unsupported queries, just those that fail prepare. The advantage of "soft" is that it is also largely non-descriptive so it does not end up being obviously inaccurate. Feel free to suggest a better name; in the mean time I'll keep "soft".
</details>
how about "allow-prepare-fail"?

---

*[sql/logic_test.go, line 192 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLH_3eqwwm0usqOhyfM:-KLJBUsYs4Y6Wd0GxPed:1849843225) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L192)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Yes, as explained in the comment above.
</details>
oh, i missed that. s/number/numeric/ here

---

*[sql/logic_test.go, line 329 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHb_bN0XB80RO89RAF:-KLJBqHncWu5Sd6opcdb:-1589215883) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L329)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Done.
</details>
indicates

---

*[sql/logic_test.go, line 147 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJBIEpnsCivD5ZZdeH:-KLJBIEqAxut9OIP-3iI:-1180748942) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L147)):*
> ```Go
> //
> // The output generated by the following flags is suppressed unless
> // either -v is given, -verbosity N with N>=1, or a test fails.
> ```

I'm pretty sure `-verbosity` won't do it.

---

*[sql/logic_test.go, line 503 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJC4YfCWa5-AGJd0jw:-KLJC4YfCWa5-AGJd0jx:1422792622) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L503)):*
> ```Go
> 	for s.Scan() {
> 		if *maxErrs > 0 && t.failures >= *maxErrs {
> 			t.Fatalf("too many errors encountered")
> ```

you don't need to emit an error, just call `t.FailNow()`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: all files reviewed at latest revision, 9 unresolved discussions, all commit checks successful.

---

*[sql/logic_test.go, line 57 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXZW3osaZjsaqik_U:-KLM4ApwyNvhECG-cxkR:1470682476) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L57)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> sure, you could argue that, but "stress" has a specific meaning in our parlance and refers to the stress utility. large-scale is an overloaded term.
</details>
So is "large-scale" here good or bad? If you think it's bad, what better term would you suggest?

---

*[sql/logic_test.go, line 131 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHYjzT2BZs8IqXiE8m:-KLM4Q58jROG_mCqDRg0:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L131)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> how about "allow-prepare-fail"?
</details>
Done.

---

*[sql/logic_test.go, line 192 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLH_3eqwwm0usqOhyfM:-KLM5SdbphGJiarnTRbs:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L192)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> oh, i missed that. s/number/numeric/ here
</details>
Done.

---

*[sql/logic_test.go, line 329 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHb_bN0XB80RO89RAF:-KLM5V_X8CWtvcKIEjOc:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L329)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> indicates
</details>
Done.

---

*[sql/logic_test.go, line 147 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJBIEpnsCivD5ZZdeH:-KLM53X9FslL5_Y62ZOr:1630992968) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L147)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> I'm pretty sure `-verbosity` won't do it.
</details>
Somehow the condition in the code was `testing.Verbose() || log.V(1)` so that's where this comment came from. I simplified to `testing.Verbose()`.

---

*[sql/logic_test.go, line 503 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJC4YfCWa5-AGJd0jw:-KLM5ju-28YJCzfhbtc_:274384708) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L503)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> you don't need to emit an error, just call `t.FailNow()`
</details>
Yes I know but it is useful for the user to know that the end of the input was not reached. I just clarified the message too.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[sql/logic_test.go, line 57 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXZW3osaZjsaqik_U:-KLMW-VxCjR9mGL_ydrj:2100527146) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L57)):*
> TestLogic implements the infrastructure that runs end-to-end tests against CockroachDB's SQL layer. It is typically used to run CockroachDB's own tests (stored in the `testdata` directory) during development and CI, and a subset of SQLite's sqllogictest during nightly CI runs. However, any test input can be specified via command-line flags (see below).

---

*[sql/logic_test.go, line 147 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJBIEpnsCivD5ZZdeH:-KLMX3TidH6NqF0sES56:905392523) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L147)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Somehow the condition in the code was `testing.Verbose() || log.V(1)` so that's where this comment came from. I simplified to `testing.Verbose()`.
</details>
I think the condition being `testing.Verbose() || log.V(1)` is fine - the point I was making is that in order for _anything_ to be printed at all, either `-v` must be specified, or a test must fail.

In other words, before your latest change, `-verbosity=1` would cause extra output **in the event of a failed test**. This is no longer true after this latest change.

---

*[sql/logic_test.go, line 502 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLMXXlpq1vyfJOE69xK:-KLMXXlpq1vyfJOE69xL:-827967676) ([raw file](https://github.com/cockroachdb/cockroach/blob/21af0f5758bf9f2c0223d6b44043ee71a2ba65d6/sql/logic_test.go#L502)):*
> ```Go
> 	for s.Scan() {
> 		if *maxErrs > 0 && t.failures >= *maxErrs {
> 			t.Fatalf("%s:%d: too many errors encountered, skipping the rest of the input")
> ```

did you mean to pass additional arguments?

---

*[sql/testdata/join, line 31 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLMXhNQr_itnjl-r7ve:-KLMXhNQr_itnjl-r7vf:1259055502) ([raw file](https://github.com/cockroachdb/cockroach/blob/21af0f5758bf9f2c0223d6b44043ee71a2ba65d6/sql/testdata/join#L31)):*
> ```
> 42 42
> 
> query I colnames label-short
> ```

:+1: though i still don't understand what the labels do :(

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 0 of 1 files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---

*[sql/logic_test.go, line 57 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLHXZW3osaZjsaqik_U:-KLNk_gjrHxuuoXcgNLL:-1834867277) ([raw file](https://github.com/cockroachdb/cockroach/blob/5b3befdd4616a2cf5ef9522db394f11ef72853f1/sql/logic_test.go#L57)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> > TestLogic implements the infrastructure that runs end-to-end tests against CockroachDB's SQL layer. It is typically used to run CockroachDB's own tests (stored in the `testdata` directory) during development and CI, and a subset of SQLite's sqllogictest during nightly CI runs. However, any test input can be specified via command-line flags (see below).
</details>
Thanks!

---

*[sql/logic_test.go, line 147 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJBIEpnsCivD5ZZdeH:-KLNkoj-LmonPp_PfV6Q:1530444403) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L147)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> I think the condition being `testing.Verbose() || log.V(1)` is fine - the point I was making is that in order for _anything_ to be printed at all, either `-v` must be specified, or a test must fail.
> 
> In other words, before your latest change, `-verbosity=1` would cause extra output **in the event of a failed test**. This is no longer true after this latest change.
</details>
Ok `|| log.V(1)` it is. Thanks for explaining.

---

*[sql/logic_test.go, line 502 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLMXXlpq1vyfJOE69xK:-KLNkxfX1kYwfMCVcnQ8:285157907) ([raw file](https://github.com/cockroachdb/cockroach/blob/21af0f5758bf9f2c0223d6b44043ee71a2ba65d6/sql/logic_test.go#L502)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> did you mean to pass additional arguments?
</details>
Woops, indeed!

---

*[sql/testdata/join, line 31 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLMXhNQr_itnjl-r7ve:-KLNl-VqaCuqCHVMBZjF:-1651512804) ([raw file](https://github.com/cockroachdb/cockroach/blob/21af0f5758bf9f2c0223d6b44043ee71a2ba65d6/sql/testdata/join#L31)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> :+1: though i still don't understand what the labels do :(
</details>
https://www.sqlite.org/sqllogictest/doc/trunk/about.wiki
"""
The <label> argument is also optional. If included, sqllogictest stores a hash of the results of this query under the given label. If the label is reused, then sqllogictest verifies that the results are the same. This can be used to verify that two or more queries in the same test script that are logically equivalent always generate the same output. 
"""

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[sql/logic_test.go, line 147 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJBIEpnsCivD5ZZdeH:-KLNmZ_K2AipJlnnBT_v:-921913030) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L147)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> Ok `|| log.V(1)` it is. Thanks for explaining.
</details>
Publish-before-push?

---

*[sql/testdata/join, line 31 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLMXhNQr_itnjl-r7ve:-KLNmQDxO640WxfqnkRB:474901735) ([raw file](https://github.com/cockroachdb/cockroach/blob/21af0f5758bf9f2c0223d6b44043ee71a2ba65d6/sql/testdata/join#L31)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary>
> https://www.sqlite.org/sqllogictest/doc/trunk/about.wiki
> """
> The <label> argument is also optional. If included, sqllogictest stores a hash of the results of this query under the given label. If the label is reused, then sqllogictest verifies that the results are the same. This can be used to verify that two or more queries in the same test script that are logically equivalent always generate the same output. 
> """
</details>
Ah, I see. Yeah, let's keep this out - it's not helpful in this case AFAICT.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks pending.

---

*[sql/logic_test.go, line 147 \[r4\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-KLJBIEpnsCivD5ZZdeH:-KLNnLCtTisueocMX4q9:-1355174333) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac6f2234bc04c5b883d5a02355ec4e036e4cfd7b/sql/logic_test.go#L147)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Publish-before-push?
</details>
I had reverted the condition earlier already. You can see it in TestLogic() on line 1110.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7409#-:-KLNpA_0SwsbIeyWYtu-:1417088721)*
<!-- Sent from Reviewable.io -->
